Use custom classloader for scripted tests

eatkins opened the pull request:
We had previously used reflection to load the bridge class, but
continued using sbt's default classloader. This was problematic because
the metabuild could have a different classpath from that required by the
scripted tests.

Bonus: scalafmt

Fixes: #4514

- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

eatkins committed:
Use custom classloader for scripted tests

We had previously used reflection to load the bridge class, but
continued using sbt's default classloader. This was problematic because
the metabuild could have a different classpath from that required by the
scripted tests.

Bonus: scalafmt

Fixes: #4514

eed3si9n commented on project/Scripted.scala:
```diff
@@ -102,38 +104,43 @@ object Scripted {
-        callback,
-      )
-    } catch { case ite: InvocationTargetException => throw ite.getCause }
+      Thread.currentThread.setContextClassLoader(loader)
```
Thanks for figuring this out.

eatkins replied:
Necessity is the mother of invention :)

eed3si9n commented:
Could you revert 67a0962b183628c962084e24e5fe8cc789bda9cf please?

eatkins commented:
reverted

eatkins committed:
Remove debug commands in scripted

These just create noise.

eatkins committed:
Allow sbt scripted tests to run in parallel

I'm not sure if this is a huge benefit or not, but it's nice to have the
option to run the scripted tests in parallel. The default behavior
should be the same as before.

eatkins commented:
Evidently this doesn't actually fix the disabled tests. It does, however, allow me to run scripted tests on my machine using older versions of sbt.

eatkins committed:
Merge branch 'develop' into scripted-classloader
